### ISSUE TITLE
talos 1.2.0a1 does not boot up to maintenance mode

### ISSUE THREAD

**Issue by sbskas** [siderolabs/talos#6119]
### Description
Booting the iso of talos get stuck after the waiting for network address and hostname to be ready.
This happens since the 1.2.0-alpha1 version (I.e. the problem occurs with alpha2 and beta0 too).
Alpha0 version boot normaly and so does version 1.1.2.
First screen shot is the boot with 1.2.0alpha0 which boot instantly to maintenance state.
Second screenshot is the boot with the 1.2.0beta0 stuck. This screeshot has been taken after waiting for 5-6 minutes.
In both case, the same vm has been booted with different iso. dhcp server was the same.

### Logs
There is no log since the apid is not reachable. However I've been able to take screenshot of beta0 booting.
![boot-1 2 0a0-ok](https://user-images.githubusercontent.com/2479966/185675677-7bc8d249-e044-4221-b515-c3ecd7a833c3.png)
![boot1 2 0b0-ko](https://user-images.githubusercontent.com/2479966/185675680-c5c0e69a-020b-4593-b738-41d2393b004b.png)

### Environment

- Talos version: 1.2.0alpha1, alpha2, beta0
- Kubernetes version: N/A
- Platform: VMWare and bare metal (iso and ova).

**Comment by githubcdr** [comment]
I can confirm this issue is same for rpi4

**Comment by sbskas** [comment]
Bissecting git commit from 1.2.0-alpha0 to 1.2.0-alpha1:
```
git bisect good
47c35dc4740cad3f758969b5d93885c9782b439a is the first bad commit
commit 47c35dc4740cad3f758969b5d93885c9782b439a
Author: Utku Ozdemir <utku.ozdemir@siderolabs.com>
Date:   Thu Jul 21 14:58:41 2022 +0200

    feat: set stable default hostname based on machine-id
    
    Use machine-id as the source for the default hostname (e.g. `talos-2gd-76y`) instead of DHCP-assigned IP (e.g. `talos-172-20-0-2`). This way, DHCP IP changes won't impact the hostname. Defaults to true for Talos version >=1.2.
    
    Closes siderolabs/talos#5896.
    
    Signed-off-by: Utku Ozdemir <utku.ozdemir@siderolabs.com>

 go.mod                                             |  1 +
 go.sum                                             |  2 +
 hack/release.toml                                  |  9 +++
 .../pkg/controllers/network/hostname_config.go     | 55 ++++++++++++++++-
 .../controllers/network/hostname_config_test.go    | 71 +++++++++++++++++++++-
 pkg/machinery/config/contract.go                   |  6 ++
 pkg/machinery/config/contract_test.go              | 25 ++++++++
 pkg/machinery/config/provider.go                   |  1 +
 .../config/types/v1alpha1/generate/init.go         |  4 ++
 .../config/types/v1alpha1/v1alpha1_features.go     |  7 +++
 .../config/types/v1alpha1/v1alpha1_types.go        |  3 +
 .../config/types/v1alpha1/v1alpha1_types_doc.go    |  7 ++-
 .../config/types/v1alpha1/zz_generated.deepcopy.go |  5 ++
 website/content/v1.2/reference/configuration.md    |  1 +
 14 files changed, 192 insertions(+), 5 deletions(-)
```

**Comment by smira** [comment]
thanks for reporting the bug, this is going to be fixed in the upcoming `1.2.0-beta.1` version
